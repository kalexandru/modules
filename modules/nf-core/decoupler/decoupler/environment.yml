channels:
- conda-forge
- bioconda
dependencies:
- conda-forge::decoupler-py=1.8.0
- conda-forge::matplotlib=3.10.0
- conda-forge::pandas=2.2.3
- pip=24.3.1
- conda-forge::scanpy=1.10.4
