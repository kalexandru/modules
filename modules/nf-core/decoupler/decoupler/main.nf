process DECOUPLER {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/7d/7d304d1b25aa80ce8c44c8d34ea45d6b4f6f50697a4effcf1b9be4a54db19928/data' :
        'community.wave.seqera.io/library/decoupler-py_matplotlib_pandas_pip_pruned:0d4681dad9987ec5' }"

    input:
    tuple val(meta), path(mat)
    path(net)

    output:
    tuple val(meta), path("*estimate__decoupler.tsv"), emit: dc_estimate
    tuple val(meta), path("*pvals__decoupler.tsv"), emit: dc_pvals
    path("versions.yml"), emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    template 'decoupler.py'

    stub:
    """
    touch mlm_estimate__decoupler.tsv
    touch mlm_pvals__decoupler.tsv
    touch versions.yml
    """
}
