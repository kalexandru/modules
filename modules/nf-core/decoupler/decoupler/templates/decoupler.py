#!/usr/bin/env python3
import os
import pandas as pd
import argparse
import shlex

os.environ["NUMBA_CACHE_DIR"] = "./tmp"

import decoupler as dc

methods = ['aucell', 'gsea', 'gsva', 'mdt', 'mlm', 'ora', 'udt',
    'ulm', 'viper', 'wmean', 'wsum']

mat = pd.read_csv("${mat}", sep="\t", index_col=0)
net = pd.read_csv("${net}", sep="\t")

# Parse external arguments passed as a string, e.g. "--min_n 1"
raw_args = "${args}"
args_list = shlex.split(raw_args)
parser = argparse.ArgumentParser()
# Define expected external arguments. Here we only define min_n; add more as needed.
parser.add_argument("--min_n", type=int, default=1, help="Minimum n value")
parsed_args, unknown = parser.parse_known_args(args_list)

# Build the arguments dictionary for decoupler
# If you need method-specific arguments, adjust accordingly.
parsedargs = {'args': {}}
parsedargs['min_n'] = parsed_args.min_n


results = dc.decouple(
    mat=mat,
    net=net,
    **parsedargs
)

for result in results:
    results[result].to_csv(result + "__decoupler.tsv", sep="\t")

## VERSIONS FILE
with open('versions.yml', 'a') as version_file:
    version_file.write('"${task.process}":' + "\\n")
    version_file.write("\tdecoupler-py: " + dc.__version__ + "\\n")