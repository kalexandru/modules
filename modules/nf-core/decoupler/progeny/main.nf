process DECOUPLER_PROGENY {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/7d/7d304d1b25aa80ce8c44c8d34ea45d6b4f6f50697a4effcf1b9be4a54db19928/data' :
        'community.wave.seqera.io/library/decoupler-py_matplotlib_pandas_pip_pruned:0d4681dad9987ec5' }"

    input:
    tuple val(meta), path(mat)
    val(organism)
    val(top_genes)

    output:
    tuple val(meta), path("progeny_mlm_estimate_decoupler.tsv") , emit: dc_estimate
    tuple val(meta), path("progeny_mlm_pvals_decoupler.tsv"), emit: dc_pvals
    tuple val(meta), path("acts_decoupler.tsv"), emit: dc_acts
    tuple val(meta), path("figures/*umap.png"), emit: dc_umap
    tuple val(meta), path("figures/*violin.png"), emit: dc_violin
    tuple val(meta), path("matrixplot_decoupler.png"), emit: dc_matrixplot
    path("versions.yml"), emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: "{}"
"""
    #!/usr/bin/env python3
    import os

    os.environ["OMNIPATH_CACHE_DIR"] = "./tmp"
    os.makedirs(os.environ["OMNIPATH_CACHE_DIR"], exist_ok=True)

    os.environ["MPLCONFIGDIR"] = "./tmp"
    os.environ["NUMBA_CACHE_DIR"] = "./tmp"

    import pandas as pd
    import matplotlib.pyplot as plt
    import scanpy as sc
    import decoupler as dc

    adata = sc.read_h5ad('${mat}')

    progeny = dc.get_progeny(organism='${organism}', top=${top_genes})

    dc.run_mlm(
        mat=adata,
        net=progeny,
        source='source',
        target='target',
        weight='weight',
        verbose=True
    )

    adata.obsm['progeny_mlm_estimate'] = adata.obsm['mlm_estimate'].copy()
    adata.obsm['progeny_mlm_pvals'] = adata.obsm['mlm_pvals'].copy()

    mlm_estimates = pd.DataFrame(adata.obsm['mlm_estimate'], index=adata.obs_names)
    mlm_pvals = pd.DataFrame(adata.obsm['mlm_pvals'], index=adata.obs_names)
    mlm_estimates.to_csv("progeny_mlm_estimate_decoupler.tsv", sep="\t")
    mlm_pvals.to_csv("progeny_mlm_pvals_decoupler.tsv", sep="\t")

    acts = dc.get_acts(adata, obsm_key='mlm_estimate')

    sc.pl.umap(
        acts,
        color=['Trail', 'louvain'],
        cmap='RdBu_r',
        vcenter=0,
        save="Trail_louvain_umap.png"
    )

    sc.pl.violin(
        acts,
        keys=['Trail'],
        groupby='louvain',
        rotation=90,
        save="Trail_violin.png"
    )

    sc.pl.matrixplot(
        acts,
        var_names=acts.var_names,
        groupby='louvain',
        dendrogram=True,
        standard_scale='var',
        colorbar_title='Z-scaled scores',
        cmap='RdBu_r'
    )
    plt.savefig("matrixplot_decoupler.png", dpi=300)

    acts_df = pd.DataFrame(acts.X, index=acts.obs_names, columns=acts.var_names)
    acts_df.to_csv("acts_decoupler.tsv", sep="\t")

    with open('versions.yml', 'a') as version_file:
        version_file.write('"${task.process}":' + "\\n")
        version_file.write("\tdecoupler-py: " + dc.__version__ + "\\n")
"""

}
