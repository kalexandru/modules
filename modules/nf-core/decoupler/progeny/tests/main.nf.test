nextflow_process {

    name "Test Process DECOUPLER_PROGENY"
    script "../main.nf"
    config "./nextflow.config"
    process "DECOUPLER_PROGENY"

    tag "modules"
    tag "modules_nfcore"
    tag "decoupler"
    tag "decoupler/progeny"

    test("decoupler_progeny_test") {

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file("https://raw.githubusercontent.com/chanzuckerberg/cellxgene/main/example-dataset/pbmc3k.h5ad", checkIfExists: true)
                ]
                input[1] = Channel.value("human")
                input[2] = Channel.value("1000")
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
