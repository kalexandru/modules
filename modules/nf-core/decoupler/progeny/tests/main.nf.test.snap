{
    "decoupler_progeny_test": {
        "content": [
            "progeny_mlm_estimate_decoupler.tsv",
            "progeny_mlm_pvals_decoupler.tsv",
            "acts_decoupler.tsv",
            "umapTrail_louvain_umap.png",
            "violinTrail_violin.png",
            "matrixplot_decoupler.png",
            [
                "versions.yml:md5,1f1551db219b78d1db7614bfa9e178d4"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-26T15:19:28.286507743"
    }
}